### PR TITLE
SymLinks cross compatibility between Apache versions

### PR DESCRIPTION
**Pull request by angelo983**
### Description (*)

This file change to solve error 500 with Apache 2.4 installed with a default configuration.

### Fixed Issues (if relevant)
Error 500 .htaccess: Option FollowSymLinks not allowed here

### Manual testing scenarios (*)

1. Apache 2.4
2. Magento 2.3

### Questions or comments

Due to security reasons Apache 2.4 comes with FollowSymLinks disabled and SymLinksIfOwnerMatch enabled:
- http://httpd.apache.org/docs/2.2/misc/perf-tuning.html#symlinks
- http://httpd.apache.org/docs/2.4/en/mod/mod_rewrite.html#RewriteRule

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by angelo983** [comment]
This pull request replaces #23545
I did this because of squashing commits, but nothing changed.
Where am I doing it wrong?

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5709](https://jira.corp.magento.com/browse/ENGCOM-5709) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@angelo983 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by ihor-sviziev** [comment]
Hi @angelo983,
Looks like you pushed your changes that not related to this PR. Could you force push your branch to latest commit that related to this PR?

File: app/code/Magento/Checkout/i18n/en_US.csv [hunk 1]
```diff
@@ -85,6 +85,8 @@ Edit,Edit
 "Click <a href=""%1"">here</a> to continue shopping.","Click <a href=""%1"">here</a> to continue shopping."
 "Estimate Tax","Estimate Tax"
 "Estimate Shipping and Tax","Estimate Shipping and Tax"
+"Enter your billing address to get a tax estimate.","Enter your billing address to get a tax estimate."
```
Comment (ihor-sviziev) [inline app/code/Magento/Checkout/i18n/en_US.csv]: These changes are not related to this PR

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by angelo983 — COMMENTED** [review]
Sometimes GitHub makes me go crazy. Please remove this commit, I will open a new Pull Request with another branch for Label not translated in Summary Estimate Shipping

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5709](https://jira.corp.magento.com/browse/ENGCOM-5709) has been created to process this Pull Request

**Review by rodrigowebjump — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, thank you for the review.
[ENGCOM-5709](https://jira.corp.magento.com/browse/ENGCOM-5709) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
@angelo983 Could you, please, provide more details how to reproduce the issue? Thanks!

**Comment by angelo983** [comment]
@engcom-Alfa Install by composer Magento 2.3 in a server with Apache 2.4 standard configuration. Due to security reasons Apache 2.4 comes with FollowSymLinks disabled and SymLinksIfOwnerMatch enabled, so without these changes to .htaccess files you will have immediatly an error in home page too.

**Comment by ihor-sviziev** [comment]
On which OS are you installing Apache? I tried official docker images with
Apache And mod php - it works fine

**Comment by angelo983** [comment]
I replayed the problem in both CentOs 7 and Debian 9 with Apache 2.4 default config with fastcgi or nginx and php 7.1 or 7.2

**Comment by ihor-sviziev** [comment]
@engcom-Alfa, could you review comment by @angelo983 and try to reproduce it again?

**Comment by engcom-Alfa** [comment]
@ihor-sviziev , ok. I will try to reproduce the issue again

**Comment by ihor-sviziev** [comment]
Hi @engcom-Alfa,
Is there any updates? This PR is in testing for 12 days already

File: .htaccess [hunk 2]
```diff
@@ -140,7 +140,12 @@
 ############################################
 ## enable rewrites
 
-    Options +FollowSymLinks
+    <IfVersion < 2.4>
+        Options +FollowSymLinks
+    </IfVersion>
+    <IfVersion >= 2.4>
+        Options +SymLinksIfOwnerMatch
```
Comment (engcom-Golf) [inline .htaccess:147]: This is not default apache configuration for all systems

**Review by engcom-Golf — CHANGES_REQUESTED** [review]
Hi, @angelo983 thank you for Pull Request, but we have some troubling with changes.  this config does not default apache configuration, it depends on distributive or Hosting provider configuration like this -> https://wiki.centos-webpanel.com/options-followsymlinks-not-allowed so The way by depending configuration by apache version not correct here. as you can see we have a few functional tests fail with these changes, as symlinks now are broken. Could you please fix the failed test and provide the solution not depending on the apache version?

**Comment by ihor-sviziev** [comment]
@angelo983  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**